### DISCUSSION TITLE
Only data is reported via kafka fetch

### DISCUSSION THREAD

**Discussion by 1450793561** [apache/skywalking#11776] [Q&A] [answered]
I just want to pass the kafka fetch report data, https://skywalking.apache.org/docs/main/v9.7.0/en/setup/backend/kafka-fetcher/, Must the agent configuration file also be configured with collector.backend_service? The current tests show that an error occurs when the collector.backend_service parameter is not configured
<img width="1067" alt="企业微信截图_2bd65a8d-b107-414c-af22-bb3399775535" src="https://github.com/apache/skywalking/assets/31468018/0f737b0a-b141-4ef6-a6d1-ae36d7f0e956">

**Comment by 1450793561** [comment]
Do kafka fetch and grpc have to exist together? What kind of data is reported through kafka?

**Reply by Superskyyy** [reply]
yes, metadata

**Comment by wu-sheng** [comment, marked as answer]
> Must the agent configuration file also be configured with collector.backend_service? The current tests show that an error occurs when the collector.backend_service parameter is not configured

Yes, you have to. Kafka only is going to support metrics, logs, and traces transport. Meanwhile, agent configuration and management commands are pushed down through gRPC only. Because the Kafka is one-way channel only.

**Comment by 1450793561** [comment]
Are kafka fetch and grpc backup methods? For example, kafka fech will report data through grpc when the connection fails. Or is an oap connection failure reported via kafka fetch?How are these two configured in best practice?

**Reply by wu-sheng** [reply]
No, there isn't. The two channels have different responsibilities as I described.
